sale_quotation/models/sale_order.py

```diff
@@ -42,29 +84,164 @@ def _compute_quotation_state(self):
+                    current_state=self.quotation_state,
+                )
+            if exception:
+                if raise_exception:
+                    raise exception
+                return False
+        return True
+
+    def action_customer_request_quotation(self):
+        self._check_customer_action_allowed("request_quotation")
+        self.quotation_state = "customer_request"
+        return True
+
+    def action_customer_accept_quotation(self):
+        self._check_customer_action_allowed("accept_quotation")
         self.quotation_state = "accepted"
+        self.typology = "sale"
+        return self.with_context(
+            bypass_customer_quotation=True,
+        ).action_confirm()
+
+    def action_customer_reset_quotation_to_draft(self):
+        self._check_customer_action_allowed("reset_to_draft")
+        self.quotation_state = "draft"
+        self.state = "draft"
+        self.filtered(lambda so: so.typology != "quote").typology = "quote"
+        return self.with_context(
+            bypass_customer_quotation=True,
+        ).action_draft()
+
+    def action_customer_cancel_quotation(self):
+        self._check_customer_action_allowed("cancel_quotation")
+        self.quotation_state = "cancel"
+        return self.with_context(
+            disable_cancel_warning=True,
+        ).action_cancel()
+
+    def action_confirm(self):
+        other_quotations = self
+        if not self.env.context.get("bypass_customer_quotation", False):
+            customer_quotations = self.filtered("use_customer_quotation_workflow")
+            if (
+                customer_quotations
+                and self.env.context.get("use_quotation_confirm_wizard")
+                and any(rec.quotation_state != "waiting_acceptation" for rec in self)
+            ):
+                return {
+                    "name": _("Confirm Sale Order"),
+                    "type": "ir.actions.act_window",
+                    "res_model": "sale.order.confirm.warning.wizard",
+                    "views": [[False, "form"]],
+                    "target": "new",
+                    "context": {
+                        "default_sale_order_ids": self.ids,
+                        "default_message": _(
+                            "The selected quotation(s) are not in 'Waiting Acceptation' "
+                            "state. Are you sure you want to confirm them?"
+                        ),
+                    },
+                }
+            else:
+                customer_quotations.action_customer_accept_quotation()
+        return super(SaleOrder, other_quotations).action_confirm()
+
+    def action_draft(self):
+        other_quotations = self
+        if not self.env.context.get("bypass_customer_quotation", False):
+            customer_quotations = self.filtered("use_customer_quotation_workflow")
+            if customer_quotations:
+                self.action_customer_reset_quotation_to_draft()
```

Comment:
Shouldn't this be:

```suggestion
                customer_quotations.action_customer_reset_quotation_to_draft()
```

?